scrapyd-client/scrapyd-deploy

```diff
@@ -52,24 +52,35 @@ def parse_opts():
         help="use the given egg, instead of building it")
     parser.add_option("--build-egg", metavar="FILE",
         help="only build the egg, don't deploy it")
+    parser.add_option("--settings-module", metavar="SCRAPY_SETTINGS_MODULE",
+        help="python module to load as scrapy settings, can be passed also "
+        "through the env var SCRAPY_SETTINGS_MODULE")
+    parser.add_option("--url", metavar="SCRAPYD_URL",
+        help="URL to the scrapyd server, overrides the url definition in the "
+        "target if any")
     return parser.parse_args()
 
 def main():
     opts, args = parse_opts()
     exitcode = 0
+    if opts.settings_module:
+        os.environ['SCRAPY_SETTINGS_MODULE'] = opts.settings_module
+
     if not inside_project():
-        _log("Error: no Scrapy project found in this location")
+        _log("Error: no Scrapy project found in this location and no url"
```

Comment:
Yes, or something like "url and settings arguments are required when not inside a project".
But still, what is your reason for working without a project?
How do you run your project's spiders locally?
We need valid use-cases before adding new features.

Also, if we are going to duplicate all scrapy.cfg options, we'd better name them the same.
E.g. `--deploy.url`, like in git.
Although there's a problem with this, the version option in the config has different semantics
from the version argument.